README.md

```diff
@@ -1,2 +1,77 @@
 # fetch
-Download and validate versioned releases of GitHub repos
+
+fetch will download a file or folder from a specific tag of a GitHub repo, subject to the [Semantic Versioning](http://semver.org/) constraints you impose.
+
+It is well-suited to downloading the latest version of a file or folder published in a GitHub repo such that you get 
+the latest non-breaking-change version. Basically, it's like a package manager, but for arbitrary GitHub repos.
+
+## Motivation
+[Gruntwork](http://gruntwork.io) helps software teams get up and running on AWS with DevOps best practices and world-class 
+infrastructure in about 2 weeks. Sometimes we publish scripts that clients use in their infrastructure, and we want clients
+to auto-download the latest non-breaking version of a script when we publish updates. In addition, for security reasons,
+we wish to verify the integrity of the git commit being downloaded.
+ 
+## Installation
+Download the binary from the [GitHub Releases](https://github.com/gruntwork-io/script-modules/releases) tab. 
+
+## Assumptions
+fetch assumes that a repo's tags are in the format `vx.y.z` or `x.y.z` to support Semantic Versioning parsing. Repos that
+use git tags not in this format cannot be used with fetch.
+
+## Usage
+
+#### General Usage
+```
+fetch --repo=<github-repo-url> --tag=<version-constraint> /repo/path/to/file/or/directory /output/path/to/file/or/directory
+```
+
+Run `fetch --help` to see more information about each argument. See [Version Constraint Operators](#version-constraint-operators)
+for examples of version constraints you can use.
+
+#### Example
+
+Download `/modules/cloudwatch-log-aggregation/install-cloudwatch-logs-agent.sh` from a GitHub tagged release where the tag is the latest version of 0.1.x but at least 0.1.5, and save it to `/tmp/cloudwatch-log-aggregation/`. 
```

Comment:
Seems like an overly specific example. Download `/foo/bar` may be a bit simpler :)